ui/store/actions.js

```diff
@@ -671,6 +671,22 @@ const updateMetamaskStateFromBackground = () => {
   });
 };
 
+export function updateSwapApprovalTransaction(txSwapApproval, dontShowLoadingIndicator) {
+  return async (dispatch) => {
+    !dontShowLoadingIndicator && dispatch(showLoadingIndication());
+
+    try {
+      await promisifiedBackground.updateSwapApprovalTransaction(txSwapApproval.id, txSwapApproval);
```

Comment:
I think it would be best if `updateSwapApprovalTransaction` had the id as the first param, and the metadata as the second param, the same as the controller method

Reply:
Also, for this method, we don't need to show the loading indicator, so you can delete the lines and params related to that

Reply:
I remove some of those line, only return is needed.